### PR TITLE
Generate Random Character

### PR DESCRIPTION
**Pull request by TheRealPikachu**
**Please describe the changes this PR makes and why it should be merged:**

**Status**

- [ ] Code changes have been tested against the Discord API and the discord.js wrapper, or there are no code changes
- [ ] Documentation has been added/modified, or there is nothing to change (docs/mods.json)

**Semantic versioning classification:**

- [ ] This PR changes DBM's interface (methods or parameters added to default methods)
- [ ] This PR **only** includes non-code changes, like changes to documentation, README, etc.

### CODE REVIEW

**Comment by RigidStudios** [comment]
I really hope this is a joke.

**Comment by TheRealPikachu** [comment]
Pretty much is

**Comment by RigidStudios** [comment]
image
So my real problem with this is the fact that your dependency (no need for one to begin with) has its own 4 dependencies, which then probably require their own dependencies.

**Comment by RigidStudios** [comment]
```js
var characters = 'ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789~!@#$%^&()*_+-={}[]';
var character = characters.charAt(math.floor(math.random() * characters.length()))
```
Instead of your obnoxious package.

File: actions/Generate Random Character [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+
+  // ---------------------------------------------------------------------
+  // Action Subtitle
+  //
+  // This function generates the subtitle displayed next to the name.
+  // ---------------------------------------------------------------------
+
+  subtitle: function (data) {
+    // Each item corresponds to each switch statement.
+    const info = ['Item 1', 'Item 2', 'Item 3']
```
Comment (RigidStudios) [inline actions/Generate Random Character:26]:
```suggestion

```

File: actions/Generate Random Character [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+    // What user sees when previewing actions box on bottom.
+    return `Generate a random symbol.`
```
Comment (RigidStudios) [inline actions/Generate Random Character:28]:
```suggestion
    return 'Generate a random character.'
```

File: actions/Generate Random Character [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+  // ---------------------------------------------------------------------
+  // Action Storage Function
+  //
+  // Stores the relevant variable info for the editor.
+  // ---------------------------------------------------------------------
+
+  variableStorage: function (data, varType) {
+    const type = parseInt(data.storage)
+    if (type !== varType) return
+    const dataType = 'Symbol'
```
Comment (RigidStudios) [inline actions/Generate Random Character:40]:
```suggestion
    const dataType = 'String'
```

File: actions/Generate Random Character [hunk 4]
```diff
@@ -0,0 +1,126 @@
...
+  // ---------------------------------------------------------------------
+  // Action Bot Mod
+  //
+  // Upon initialization of the bot, this code is run. Using the bot's
+  // DBM namespace, one can add/modify existing functions if necessary.
+  // In order to reduce conflictions between mods, be sure to alias
+  // functions you wish to overwrite.
+  // ---------------------------------------------------------------------
+
+  mod: function (DBM) {}
```
Comment (RigidStudios) [inline actions/Generate Random Character:125]:
```suggestion
  mod () {}
```
Object shorthand.

File: actions/Generate Random Character [hunk 5]
```diff
@@ -0,0 +1,126 @@
...
+
+  // ---------------------------------------------------------------------
+  // Action Bot Function
+  //
+  // This is the function for the action within the Bot's Action class.
+  // Keep in mind event calls won't have access to the "msg" parameter,
+  // so be sure to provide checks for variable existance.
+  // ---------------------------------------------------------------------
+
+  action: function (cache) {
```
Comment (RigidStudios) [inline actions/Generate Random Character:104]:
```suggestion
  action (cache) {
```
Object shorthand

File: actions/Generate Random Character [hunk 6]
```diff
@@ -0,0 +1,126 @@
...
+
+  // ---------------------------------------------------------------------
+  // Action Editor Init Code
+  //
+  // When the HTML is first applied to the action editor, this code
+  // is also run. This helps add modifications or setup reactionary
+  // functions for the DOM elements.
+  // ---------------------------------------------------------------------
+
+  init: function () {},
```
Comment (RigidStudios) [inline actions/Generate Random Character:94]:
```suggestion
  init () {},
```

File: actions/Generate Random Character [hunk 7]
```diff
@@ -0,0 +1,126 @@
...
+  // for an event. Due to their nature, events lack certain information,
+  // so edit the HTML to reflect this.
+  //
+  // The "data" parameter stores constants for select elements to use.
+  // Each is an array: index 0 for commands, index 1 for events.
+  // The names are: sendTargets, members, roles, channels,
+  //                messages, servers, variables
+  // ---------------------------------------------------------------------
+
+  html: function (isEvent, data) {
```
Comment (RigidStudios) [inline actions/Generate Random Character:70]:
```suggestion
  html (isEvent, data) {
```

File: actions/Generate Random Character [hunk 8]
```diff
@@ -0,0 +1,126 @@
...
+    return `Generate a random symbol.`
+  },
+
+  // ---------------------------------------------------------------------
+  // Action Storage Function
+  //
+  // Stores the relevant variable info for the editor.
+  // ---------------------------------------------------------------------
+
+  variableStorage: function (data, varType) {
```
Comment (RigidStudios) [inline actions/Generate Random Character:37]:
```suggestion
  variableStorage (data, varType) {
```

**Comment by RigidStudios** [comment]
Check my comments plskthx

**Comment by TheRealPikachu** [comment]
yeah shorthands

**Comment by almeidx** [comment]
What differs this action from [this](https://github.com/dbm-network/mods/blob/master/actions/randomize_letters_MOD.js)?

**Comment by TheRealPikachu** [comment]
holy shoot

**Pull request closed without merging**